core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -116,4 +133,21 @@ public String getMobileNetworkSubtypeName() {
         return networkInfo.getSubtypeName();
     }
 
+    /**
+     * Detects if client is behind a captive portal.
+     * @param pingerCallback to call with success or failure.
+     */
+    public void isCaptivePortal(EndpointPinger.PingerCallback pingerCallback){
+        pinger.ping(pingerCallback);
```

Comment:
Are you sure we have to do a ping? I thought that we have to check if the response is `204` because a ping can return but with a different response and so you still are behind a captive portal...

Reply:
Yes I'm sure. We use this https://github.com/novoda/merlin/blob/release/core/src/main/java/com/novoda/merlin/ResponseCodeValidator.java#L10 which checks for a `204`. In retrospect this method is a bit misleading 🤔 We will change the name and perhaps use a different interface.